### PR TITLE
Add framework for remote claims storage plugins

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
For now this just returns a filesystem claims store but later we can use this to return stores that are backed by plugins.

# What issue does it fix
Closes # _(issue)_

# Notes for the reviewer
_Put any questions or notes for the reviewer here._

# Checklist
- [ ] Unit Tests
- [ ] Documentation
  - [x] Documentation Not Impacted

### CODE REVIEW

**Comment by carolynvs** [comment]
Oops rebasing...

**Comment by carolynvs** [comment]
Oops, I need to do more work to make the integration tests work when they call porter for interal plugins. BRB rethinking my life choices 💨

File: pkg/instance-storage/claimstore/plugin.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	"github.com/deislabs/cnab-go/utils/crud"
+	"github.com/hashicorp/go-plugin"
+)
+
+const PluginKey = "instance-store"
+
+var _ plugin.Plugin = &Plugin{}
+
+// P
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply pkg/instance-storage/claimstore/plugin.go]: 😅

File: pkg/instance-storage/provider/provider.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	"github.com/deislabs/cnab-go/utils/crud"
+
+	"github.com/deislabs/porter/pkg/config"
+	instancestorage "github.com/deislabs/porter/pkg/instance-storage"
+	"github.com/deislabs/porter/pkg/instance-storage/claimstore"
+	"github.com/deislabs/porter/pkg/plugins"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	_ "github.com/hashicorp/go-plugin"
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply pkg/instance-storage/provider/provider.go]: Oops, this was when I was first working with the lib, it's not needed anymore. I'll remove it.

File: pkg/porter/helpers.go [hunk 3]
```diff
@@ -40,10 +39,13 @@ func NewTestPorter(t *testing.T) *TestPorter {
 	tc := config.NewTestConfig(t)
 	p := New()
 	p.Config = tc.Config
-	p.CNAB = cnabprovider.NewRuntime(tc.Config)
+	p.CNAB = cnabprovider.NewRuntime(tc.Config, p.InstanceStorage)
 	p.Mixins = &mixin.TestMixinProvider{}
 	p.Cache = cache.New(tc.Config)
+
 	p.Builder = NewTestBuildProvider()
+	p.InstanceStorage = instancestorage.NewTestInstanceStorageProvider()
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply pkg/porter/helpers.go]: It does! I'm fixing it now locally to get the integration tests working

**Comment by carolynvs** [comment]
Sorry about the low quality PR. I should have waited on asking for feedback until I had figured out why the integration test for dependencies was failing first.

**Comment by vdice** [comment]
@carolynvs Your PRs are the highest-quality around!  I always learn a ton from them, both code-related and how to make better PRs myself :)

**Pull request merged**